### PR TITLE
UCT/IB: added an option to set whether or not PFC is enabled on the switch.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2330/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1662/ for details (Mellanox internal link).

**Comment by alinask** [comment]
@yosefe please review.

File: src/uct/ib/base/ib_md.c [hunk 1]
```diff
@@ -42,6 +42,14 @@ static ucs_config_field_t uct_ib_md_config_table[] = {
    "Initialize a fork-safe IB library with ibv_fork_init().",
    ucs_offsetof(uct_ib_md_config_t, fork_init), UCS_CONFIG_TYPE_TERNARY},
 
+  {"PFC_ENABLED", "n",
+   "Whether or not PFC (Pause Frame Control) is enabled on the switch. \n"
+   "A mechanism for temporarily stopping the transmission of data to \n"
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c]: PFC is a ...

File: src/uct/ib/base/ib_md.c [hunk 2]
```diff
@@ -42,6 +42,14 @@ static ucs_config_field_t uct_ib_md_config_table[] = {
...
+   "ensure zero loss under congestion on Ethernet family computer networks. \n"
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c]: pls remove " " before "\n"

**Comment by yosefe** [comment]
@alexvm can you please take a look?

**Comment by alexvm** [comment]
>>Whether or not PFC (Pause Frame Control)
Do you mean: Priority-based flow control?

>> is enabled on the switch
PFC should be enabled also on the NIC

**Comment by yosefe** [comment]
@alinask let's change it to `ETH_PAUSE_ON=[y|n]` and the doc to "Whether or not PAUSE frame is enabled on Ethernet network"
https://en.wikipedia.org/wiki/Ethernet_flow_control

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2340/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1673/ for details (Mellanox internal link).

File: src/uct/ib/base/ib_md.h [hunk 3]
```diff
@@ -55,6 +56,8 @@ typedef struct uct_ib_md_config {
                                                 without using the cache */
 
     unsigned                fork_init;     /**< Use ibv_fork_init() */
+    int                     pfc_enabled;   /**< Whether or not PFC is enabled on the switch
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.h]: this should be also renamed accordingly

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2344/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1678/ for details (Mellanox internal link).

**Comment by alinask** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2354/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1687/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2358/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1693/ for details (Mellanox internal link).

**Comment by alinask** [comment]
bot:mlx:retest

**Comment by Di0gen** [comment]
bot:mlx:retest

**Comment by alinask** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2362/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1697/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by alinask** [comment]
@MattBBaker could you please post the ORNL failure here as well?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2369/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1706/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
:+1:

**Comment by MattBBaker** [comment]
@alinask Failure here was same as in #963, changes in the test system caused gcc to be unhappy about a compare between signed/unsigned.

**Pull request merged**